Moved calendar onKeyDown logic to slate plugin

karenzheung opened the pull request:
_Pull requests into Flux require the following. Submitter should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable (**N/A**) and :white_check_mark:. Reviewers should complete every item on the bulleted list when reviewing._


## Submitter:

This addresses jira 2125. Calendar no longer steals focus from editor and added onKeyDown logic to the calendar. 

**Running the Application**

- [x] Manually tested in Chrome (primary browser for testing)
- [ ] Manually tested in IE/Safari (verify app loads and basic feature testing)
- [ ] Manually tested in Chrome in iPad mode (at minimum, viewing patient record)

**Task**

- [x] Branch implements task as expected
- [x] Key application features are updated as needed for task and tested (see: [Key Application Features](https://github.com/FluxNotes/flux/wiki/Key-Application-Features))
- [x] Task specific definition of done (as described in the Jira task) is met, if applicable.

**Documentation**

- [x] This pull request describes why these changes were made
- [x] Recognizes any potential shortcomings/bugs in the description 
- [ ] Pre release script is updated on the [Pre Release Testing Script wiki page](https://github.com/FluxNotes/flux/wiki/Pre-Release-Testing-Script) (add/remove single features to test, do not run through full script)
- [ ] Update the list of key features with new features on the [Key Application Features wiki page](https://github.com/FluxNotes/flux/wiki/Key-Application-Features)
- [ ] Document any new APIs/extension points
- [ ] Additional technical components and libraries are documented and added to [wiki](https://github.com/FluxNotes/flux/wiki/About-Flux-Notes) as needed

**Code Quality**

- [x] Code style guide is followed (see: [Code Style Guide](https://github.com/FluxNotes/flux/wiki/JavaScript-and-HTML-Code-Style-Guide))
- [x] Code is commented

**Tests**

- [x] Existing tests passed
- [x] Added backend tests for new functionality added


## Reviewers:

- Check that code is maintainable and reusable. Code reuses existing code and infrastructure where appropriate.
- Check the PR and code accomplishes the task's purpose.
- Check that new tests appropriately test the new code, including edge cases. Other existing tests pass.
- Verify Pull Request checklist is correctly completed by submitter.
- Try to break the code

Dtphelan1 commented on src/context/ContextCalendar.jsx:
```diff
@@ -9,12 +16,45 @@ class ContextCalendar extends React.Component {
+            const date = this.cal.state.selectedValue;
+            const context = { key: 'set-date-id', context: `${date}`, object: date };
+            // If a plugin returns a state, it short circuits future plugins
+            return this.props.onSelected(this.props.state, context);
```
[DP3?] Instead of replicating the functionality in handleDateSelect, is there any way we could call handleDateSelect directly? It looks like there is a subtle difference in that the original function didn't have a return value, but I'm wondering if anything would go awry if we added a return in that function. Could we explore?

karenzheung commented:
Addressed DP1 and DP3. 

For DP2 I removed the setTimeout. The setTimeout was there because previously the calendar was not always being focused but since this approach doesn't have input.focus I do not think it's necessary.

mgramigna commented on src/notes/FluxNotesEditor.jsx:
```diff
@@ -87,7 +87,7 @@ class FluxNotesEditor extends React.Component {
     closeCompletionPortal = () => {
         if (this.state.completionComponentShortcut) {
-            // Clean up some variables stored at the Editor level
+
```
[MG1] Any reason for the removal of this comment? If not can we add it back?

mgramigna commented on src/context/ContextCalendar.jsx:
```diff
@@ -9,12 +16,41 @@ class ContextCalendar extends React.Component {
+        }
+    }
+
+
```
[MG2] Can we move the calls to preventDefault and stopPropagation to a function so we don't have to keep calling them individually?

karenzheung commented:
Resolved MG1+2

Dtphelan1 commented:
@mgramigna  re-review when you can?

mgramigna commented:
[MG3] Can't hit enter to select anymore after the refactor :/

karenzheung commented:
Looking into MG3 now

mgramigna commented:
If it's some weirdness about moving the event stopping to a function due to non-obvious weirdness you can just revert MG2

karenzheung commented:
Fixed MG3. The reason hitting enter stopped working was because I wasn't returning the state from this.handleDateSelect in the onKeyDown.

mgramigna commented:
Will merge after CI
